Sources/Sentry/SentryNSDataTracker.m

```diff
@@ -173,9 +207,43 @@ - (NSData *)measureNSDataFromURL:(NSURL *)url
 
     [ioSpan setDataValue:path forKey:@"file.path"];
 
+    [self mainThreadExtraInfo:ioSpan];
+
     return ioSpan;
 }
 
+- (void)mainThreadExtraInfo:(id<SentrySpan>)span
+{
+    BOOL isMainThread = [NSThread isMainThread];
+
+    [span setDataValue:@(isMainThread) forKey:@"blocked_main_thread"];
+
+    if (!isMainThread) {
+        return;
+    }
+
+    SentryThreadInspector *threadInspector = self.threadInspector;
+    SentryStacktrace *stackTrace = [threadInspector stacktraceForCurrentThreadNatively];
+
+    NSMutableArray *frames = [[NSMutableArray alloc] initWithCapacity:stackTrace.frames.count];
+    for (SentryFrame *frame in stackTrace.frames) {
+        // We dont need the frame if is a system frame
+        if ([frame.package hasPrefix:self.processInfoWrapper.processDirectoryPath]) {
+            [frames addObject:[frame serialize]];
+        }
+    }
+
+    if (frames.count <= 1) {
+        // This means the call was made only by system APIs,
+        // therefore, there is nothing to do about it
+        // and we should not report it as an issue.
+        [span setDataValue:@(NO) forKey:@"blocked_main_thread"];
```

Comment:
`m`: In that case, I wouldn't set anything. I would only set `blocked_main_thread` if the span blocked the main thread and the frames contain non-system frames.

Reply:
Same as https://github.com/getsentry/sentry-cocoa/pull/2573/files#r1071058845